agnhost sync

BenTheElder opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #92059 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

BenTheElder committed:
bump agnhost usage to current version

BenTheElder committed:
update agnhost guestbook usage for s/slave/replica/

BenTheElder commented:
follow-up to #87105

BenTheElder commented:
blocked on https://github.com/kubernetes/k8s.io/pull/958

BenTheElder commented:
/retest

On Mon, Jun 15, 2020, 21:53 Kubernetes Prow Robot <notifications@github.com>
wrote:

> @BenTheElder <https://github.com/BenTheElder>: The following tests
> *failed*, say /retest to rerun all failed tests:
> Test name Commit Details Rerun command
> pull-kubernetes-node-e2e d65a3b1
> <https://github.com/kubernetes/kubernetes/commit/d65a3b15bf4a720b09c32e9fff06928aab43c9e5>
> link
> <https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/92119/pull-kubernetes-node-e2e/1272727789052628992/> /test
> pull-kubernetes-node-e2e
> pull-kubernetes-integration d65a3b1
> <https://github.com/kubernetes/kubernetes/commit/d65a3b15bf4a720b09c32e9fff06928aab43c9e5>
> link
> <https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/92119/pull-kubernetes-integration/1272727789035851779> /test
> pull-kubernetes-integration
> pull-kubernetes-conformance-kind-ipv6-parallel d65a3b1
> <https://github.com/kubernetes/kubernetes/commit/d65a3b15bf4a720b09c32e9fff06928aab43c9e5>
> link
> <https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/92119/pull-kubernetes-conformance-kind-ipv6-parallel/1272727789027463168/> /test
> pull-kubernetes-conformance-kind-ipv6-parallel
> pull-kubernetes-e2e-kind d65a3b1
> <https://github.com/kubernetes/kubernetes/commit/d65a3b15bf4a720b09c32e9fff06928aab43c9e5>
> link
> <https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/92119/pull-kubernetes-e2e-kind/1272727789052628993> /test
> pull-kubernetes-e2e-kind
> pull-kubernetes-e2e-kind-ipv6 d65a3b1
> <https://github.com/kubernetes/kubernetes/commit/d65a3b15bf4a720b09c32e9fff06928aab43c9e5>
> link
> <https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/92119/pull-kubernetes-e2e-kind-ipv6/1272727789044240386> /test
> pull-kubernetes-e2e-kind-ipv6
> pull-kubernetes-conformance-kind-ga-only-parallel d65a3b1
> <https://github.com/kubernetes/kubernetes/commit/d65a3b15bf4a720b09c32e9fff06928aab43c9e5>
> link
> <https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/92119/pull-kubernetes-conformance-kind-ga-only-parallel/1272727789044240388> /test
> pull-kubernetes-conformance-kind-ga-only-parallel
> pull-kubernetes-conformance-image-test d65a3b1
> <https://github.com/kubernetes/kubernetes/commit/d65a3b15bf4a720b09c32e9fff06928aab43c9e5>
> link
> <https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/92119/pull-kubernetes-conformance-image-test/1272727789031657472/> /test
> pull-kubernetes-conformance-image-test
> pull-kubernetes-e2e-gce-ubuntu-containerd d65a3b1
> <https://github.com/kubernetes/kubernetes/commit/d65a3b15bf4a720b09c32e9fff06928aab43c9e5>
> link
> <https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/92119/pull-kubernetes-e2e-gce-ubuntu-containerd/1272727789065211905/> /test
> pull-kubernetes-e2e-gce-ubuntu-containerd
> pull-kubernetes-e2e-gce d65a3b1
> <https://github.com/kubernetes/kubernetes/commit/d65a3b15bf4a720b09c32e9fff06928aab43c9e5>
> link
> <https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/92119/pull-kubernetes-e2e-gce/1272727789027463169/> /test
> pull-kubernetes-e2e-gce
>
> Full PR test history
> <https://prow.k8s.io/pr-history?org=kubernetes&repo=kubernetes&pr=92119>. Your
> PR dashboard
> <https://prow.k8s.io/pr?query=is%3Apr%20state%3Aopen%20author%3BenTheElder>.
> Please help us cut down on flakes by linking to
> <https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests>
> an open issue
> <https://github.com/kubernetes/kubernetes/issues?q=is:issue+is:open> when
> you hit one in your PR.
>
> Instructions for interacting with me using PR comments are available here
> <https://git.k8s.io/community/contributors/guide/pull-requests.md>. If
> you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://go.k8s.io/bot-commands>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/92119#issuecomment-644529844>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAHADK4E4EDQPMSO267S5CLRW33C5ANCNFSM4N5IQUOA>
> .
>

BenTheElder commented:
/retest

BenTheElder commented:
/assign @spiffxp @dims

BenTheElder commented:
/retest

dims commented:
/assign @spiffxp @johnbelamaric

BenTheElder commented:
/assign johnbelamaric
This is a simple text substitution to catch up to redis's current, less offensive terminology.
The image change has been in, but the e2e tests have fallen behind the image.

johnbelamaric commented:
Is `master` still the terminology for Redis? Should we use `primary`?

/approve

BenTheElder commented:
master appears to still redis terminology, though this is only mildly redis-like...

I think we could change it but at the time I was focused on the "API" changes to the image that couldn't be done with a simple PR.

I'll double check what changing that would entail,
/hold

We do also still have some actual extremely dated redis stuff, but I don't think any of that is in conformance and I can approve most (all?) of it myself.

BenTheElder committed:
s/master/primary in agnhost guestbook test references & agnhost code

to do in follow-up: update the agnhost deployments and references to those when the new image is pushed

BenTheElder commented:
... turns out that's going to require API changes too.

rolling up the first part of that here, since it also touches the conformance test descriptions
will file a follow-up to upgrade the image again and update the deployments again

BenTheElder commented on test/conformance/testdata/conformance.yaml:
```diff
@@ -1397,11 +1397,11 @@
-    the target port is set to 6379, port that agnhost master is listening.
+    to 6379, port that agnhost primary is listening. Using kubectl expose the agnhost
+    primary as a service at port 2345. The service MUST be listening on port 2345 and
+    the target port is set to 6379, port that agnhost primary is listening.
```
This was done by hand, there are other references to "master" in the conformance tests that are talking about:
```
$ kubectl cluster-info
Kubernetes master is running at https://127.0.0.1:42939
KubeDNS is running at https://127.0.0.1:42939/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
```

Which is a very different change. 

_some_ of those are really referring to the `kubernetes.default` / apiserver service as the "master" service (not `kubectl cluster-info` output) and could be changed trivially, but I think that's a different PR.

BenTheElder commented on test/e2e/kubectl/kubectl.go:
```diff
@@ -343,12 +343,12 @@ var _ = SIGDescribe("Kubectl client", func() {
 			for _, gbAppFile := range []string{
-				"agnhost-slave-service.yaml",
+				"agnhost-replica-service.yaml",
 				"agnhost-master-service.yaml",
```
I've already drafted the changes to these for after the e2e image is pushed

spiffxp commented:
/lgtm

spiffxp commented:
/lgtm

BenTheElder commented:
/retest

On Fri, Jun 19, 2020, 14:09 Kubernetes Prow Robot <notifications@github.com>
wrote:

> @BenTheElder <https://github.com/BenTheElder>: The following test *failed*,
> say /retest to rerun all failed tests:
> Test name Commit Details Rerun command
> pull-kubernetes-kubemark-e2e-gce-big 279ce5d
> <https://github.com/kubernetes/kubernetes/commit/279ce5d89e97a4bed8172c9e011abe8b95c92ad2>
> link
> <https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/92119/pull-kubernetes-kubemark-e2e-gce-big/1274060384617107456/> /test
> pull-kubernetes-kubemark-e2e-gce-big
>
> Full PR test history
> <https://prow.k8s.io/pr-history?org=kubernetes&repo=kubernetes&pr=92119>. Your
> PR dashboard
> <https://prow.k8s.io/pr?query=is%3Apr%20state%3Aopen%20author%3BenTheElder>.
> Please help us cut down on flakes by linking to
> <https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests>
> an open issue
> <https://github.com/kubernetes/kubernetes/issues?q=is:issue+is:open> when
> you hit one in your PR.
>
> Instructions for interacting with me using PR comments are available here
> <https://git.k8s.io/community/contributors/guide/pull-requests.md>. If
> you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://go.k8s.io/bot-commands>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/92119#issuecomment-646861338>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAHADK6NCXHIDTHUEAZ4UXLRXPHYDANCNFSM4N5IQUOA>
> .
>

BenTheElder commented:
/retest

BenTheElder commented:
/hold cancel

BenTheElder commented:
/wg naming
